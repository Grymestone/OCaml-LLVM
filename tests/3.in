int main() {
    int a = 5
    printf(a)
    0
}
