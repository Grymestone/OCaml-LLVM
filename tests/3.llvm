; ModuleID = 'tests/3.llvm'
source_filename = "awesome jit"

@buf = global [4 x i8] c"%d\0A\00"

declare i32 @printf(i8*, ...)

; Function Attrs: noinline nounwind optnone sspstrong uwtable
define i32 @main() #0 {
entry:
  %0 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @buf, i32 0, i32 0), i32 5)
  ret i32 0
}

attributes #0 = { noinline nounwind optnone sspstrong uwtable }
