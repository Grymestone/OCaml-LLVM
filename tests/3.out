VarDecExpr: int a = 5 

Setting variable a

